src/MetaMorpho.sol

```diff
@@ -284,6 +285,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         }
     }
 
+    function disableMarket(Id id) external onlyCuratorRole {
+        if (config[id].removableAt != 0) revert ErrorsLib.AlreadySet();
+        if (config[id].cap == 0) revert ErrorsLib.MarketNotEnabled();
```

Comment:
This last line has non-trivial implications: if the curator by mistake reacts fast and sets the cap of a market to zero, they can't disable it. They need to first set a non-zero cap, wait for the timelock, accept the cap then submit the market's removal and wait for the timelock again

Reply:
We could:
1. acknowledge this
2. remove this line and not set `cap = 0` upon market removal and expect the curator to do this step on their own during the timelock
3. not require `cap == 0` but still enforce it for protection

Reply:
I'm in favor of option 3.

The current implementation can be very problematic I think

Reply:
Actually option 3 is weird because if the guardian revokes the removal of the market, the cap would be left with 0 (we can't recover the previous cap)
I'd argue to expect the curator to bundle the cap set to 0 and the market's removal instead, so that each function is independent

Reply:
Arg but if we remove it we can submit a non-enabled market for removal

Reply:
@MerlinEgalite :
> why don't we do the opposite ie we require that the cap is 0?

This gave me a better idea: require the `enabled` flag, see latest changes